### PR TITLE
chore: modernize README, add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -11,11 +11,20 @@ env:
...
+    name: main - Style & Lint
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
 
+      - name: Verify README generation
+        uses: momentohq/standards-and-practices/github-actions/oss-readme-template@gh-actions-v2
+        with:
+          project_status: official
+          project_stability: alpha
```
Comment (nand4011) [inline .github/workflows/on-pull-request.yml:23]: This is beta in the generate readme section.

File: .github/workflows/on-pull-request.yml [hunk 2]
```diff
@@ -44,29 +50,27 @@ jobs:
...
         # This script relies on dev dependencies such as Tokio, so we run it with the --example flag
         run: cargo run --example test-teardown
 
   rustfmt-build-examples:
-    name: Style & Lint
+    name: examples - Style & Lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (nand4011) [inline .github/workflows/on-pull-request.yml:73]: Should we upgrade the other uses of this to v3?

File: .github/workflows/on-pull-request.yml (hunk 1 above)
Comment (cprice404) [reply .github/workflows/on-pull-request.yml:23]: oops. will change it to alpha.

File: .github/workflows/on-pull-request.yml (hunk 2 above)
Comment (cprice404) [reply .github/workflows/on-pull-request.yml:73]: sure ty

**Review by nand4011 — APPROVED** [review]

**Pull request merged**